magma/types/ready_valid.py

```diff
@@ -45,6 +45,13 @@ def __getitem__(cls, T: Union[Type, MagmaProtocol]):
     def flip(cls):
         raise TypeError("Cannot flip an undirected ReadyValid type")
 
+    def qualify(cls, direction):
```

Comment:
a general question: do these types get cached? for any of the ready-valids

Reply:
The Product metaclass handles caching for normal types, we should investigate whether there's any undesired interasctions between Products with the same field names/types as readyvalid (i.e. i think the caching should take the class into account, but it may currently only be tired to field names/types)